File: backend/app/schemas/pokemon.py
from pydantic import BaseModel, ConfigDict


class PokemonRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    pokedex_number: int
    is_default: bool
    sprite_url: str
    types: list[str]
    hp: int
    attack: int
    defense: int
    special_attack: int
    special_defense: int
    speed: int


class MoveRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    type: str
    damage_class: str
    power: int | None
    accuracy: int | None
    pp: int | None
    priority: int
    effect_chance: int | None
    effect_text: str | None


class PokemonDetail(PokemonRead):
    learnable_moves: list[MoveRead]
    type_effectiveness: dict[str, float]


class PokemonCatalogVersion(BaseModel):
    version: str | None

File: frontend/src/types/pokemon.ts
import type { Move } from "./move";

export interface Pokemon {
  id: number;
  name: string;
  pokedex_number: number;
  is_default: boolean;
  is_battle_only: boolean;
  sprite_url: string;
  types: string[];
  hp: number;
  attack: number;
  defense: number;
  special_attack: number;
  special_defense: number;
  speed: number;
}

export interface PokemonDetail extends Pokemon {
  learnable_moves: Move[];
  type_effectiveness: Record<string, number>;
}
